subsys/fs/nvs/nvs.c

```diff
@@ -399,17 +399,25 @@ static int nvs_ate_cmp_const(const struct nvs_ate *entry, uint8_t value)
 }
 
 /* nvs_ate_valid validates an ate:
- *     return 1 if crc8 and offset valid,
+ *     return 1 if crc8, offset and length are valid,
  *            0 otherwise
  */
 static int nvs_ate_valid(struct nvs_fs *fs, const struct nvs_ate *entry)
 {
 	size_t ate_size;
+	uint32_t position;
 
 	ate_size = nvs_al_size(fs, sizeof(struct nvs_ate));
+	position = entry->offset + entry->len;
 
+	/*
+	 * Subtract 3 ATEs from the sector size to take into account the ATE itself,
+	 * the sector closing ATE and the sector deletion ATE (both are always present).
+	 * However, the "GC done" ATE is not taken into account because it might not
+	 * be present.
+	 */
 	if ((nvs_ate_crc8_check(entry)) ||
-	    (entry->offset >= (fs->sector_size - ate_size))) {
+	    (position >= (fs->sector_size - (3 * ate_size)))) {
```

Comment:
@RICCIARDI-Adrien be careful here, the close ATE will become here a non valid ATE.
I don't thing that this could work

Reply:
Thanks for the review !
You're correct, the ATE reserved for deletion is kept at the end of the ATEs list, not at the beginning of the sector.
We also have to consider that the GC done ATE might be missing.
Then, if a single chunk of data occupying exactly the whole sector data area is written, its ATE may be stored just before the close ATE.
I then removed the wrong `3x` multiplier.